conans/test/unittests/tools/env/test_env.py

```diff
@@ -0,0 +1,212 @@
+import os
+import platform
+import subprocess
+import textwrap
+
+import pytest
+
+from conan.tools.env import Environment
+from conan.tools.env.environment import ProfileEnvironment
+from conans.client.tools import chdir
+from conans.test.utils.test_files import temp_folder
+from conans.util.files import save
+
+
+def test_compose():
+    env = Environment()
+    env.define("MyVar", "MyValue")
+    env.define("MyVar2", "MyValue2")
+    env.define("MyVar3", "MyValue3")
+    env.define("MyVar4", "MyValue4")
+    env.unset("MyVar5")
+
+    env2 = Environment()
+    env2.define("MyVar", "MyNewValue")
+    env2.append("MyVar2", "MyNewValue2")
+    env2.prepend("MyVar3", "MyNewValue3")
+    env2.unset("MyVar4")
+    env2.define("MyVar5", "MyNewValue5")
+
+    env3 = env.compose(env2)
+    assert env3.value("MyVar") == "MyNewValue"
+    assert env3.value("MyVar2") == 'MyValue2 MyNewValue2'
+    assert env3.value("MyVar3") == 'MyNewValue3 MyValue3'
+    assert env3.value("MyVar4") == ""
+    assert env3.value("MyVar5") == 'MyNewValue5'
+
+
+def test_profile():
+    myprofile = textwrap.dedent("""
+        # define
```

Comment:
Is it possible to specify a separator in the profile?

Reply:
No, it seems the syntax will be weird, especially to designate whitespaces